server/routes/auth/register.route.js

```diff
@@ -0,0 +1,47 @@
+const express = require("express");
+const router = express.Router();
+const bcrypt = require("bcryptjs");
+
+const validateRegisterInput = require("../../validation/auth/register.validation");
+const User = require("../../models/User");
+
+// @route POST /auth/register
+// @desc Register user
+// @access Public
+router.post("/", (req, res) => {
+  // Form validation
+  const { errors, isValid } = validateRegisterInput(req.body);
+
+  // Check validation
+  if (!isValid) {
+    return res.status(400).json(errors);
+  }
+
+  User.findOne({ email: req.body.email }).then(user => {
+    if (user) {
+      return res.status(400).json({ email: "Email already exists" });
+    } else {
+      const newUser = new User({
+        name: req.body.name,
+        email: req.body.email,
+        password: req.body.password
+      });
+      // Hash password before saving in database
+      bcrypt.genSalt(10, (err, salt) => {
+        bcrypt.hash(newUser.password, salt, (err, hash) => {
+          if (err) throw err;
+          newUser.password = hash;
+          newUser
+            .save()
+            .then(user => {
+              delete user.password; // remove hashed password from response
+              return res.json(user);
```

Comment:
For signup, we will also want to generate a JWT, so we can login the user right away.